openai/api_requestor.py

```diff
@@ -149,6 +151,63 @@ def format_app_info(cls, info):
             str += " (%s)" % (info["url"],)
         return str
 
+    def _check_polling_response(self, response: OpenAIResponse, predicate: Callable[[OpenAIResponse], bool]):
+        if predicate(response):
+            message, code = (None, None) if 'error' not in response.data else (
+                response.data['error']['message'] if 'message' in response.data['error'] else None,
+                response.data['error']['code'] if 'code' in response.data['error'] else None
+            )
+            message = message if message is not None else 'Operation failed'
+            raise error.OpenAIError(message=message, code=code)
+
+    def _poll(
+        self,
+        method,
+        url,
+        until,
+        failed,
+        params = None,
+        headers = None,
+        interval = None,
+        delay = None
+    ) -> Tuple[Iterator[OpenAIResponse], bool, str]:
+        if delay:
+            time.sleep(delay)
+
+        response, b, api_key = self.request(method, url, params, headers)
+        self._check_polling_response(response, failed)
+        while not until(response):
```

Comment:
There's a chance for an infinite loop here unless the Azure APIs guarantee that the `status` of a request always goes into either `succeeded` or `failed`. If you confirm that guarantee then this is fine

Reply:
We should add a timeout to the API/loop - but I can confirm that all long running/polling Azure APIs are supposed to end up in one of (potentially) three states (also known as terminal states in Azure) - succeeded, failed or cancelled (the latter is only for cancellable APIs).

Reply:
added a timeout (using the value already defined in api_requestor.py)